### PR TITLE
CB-20677 Unblock FEPOC customer by making sure that every resource is tagged - enforce aws native for single AZ setup

### PR DESCRIPTION
**Pull request by horadla23**
See detailed description in the commit message.

### CODE REVIEW

File: core/src/main/java/com/sequenceiq/distrox/v1/distrox/converter/DistroXV1RequestToStackV4RequestConverter.java [hunk 1]
```diff
@@ -112,6 +117,10 @@ public StackV4Request convert(DistroXV1Request source) {
         request.setExternalDatabase(getIfNotNull(source.getExternalDatabase(), databaseRequestConverter::convert));
         request.setEnableLoadBalancer(source.isEnableLoadBalancer());
         request.setVariant(source.getVariant());
+        if (CloudPlatform.AWS.equals(source.getCloudPlatform()) &&
+                entitlementService.awsNativeSingleAzDatahubEnabled(ThreadBasedUserCrnProvider.getAccountId())) {
```
Comment (doktoric) [inline core/src/main/java/com/sequenceiq/distrox/v1/distrox/converter/DistroXV1RequestToStackV4RequestConverter.java:121]: should we say **or multiazentitlementGranted()**

File: datalake/src/main/java/com/sequenceiq/datalake/service/sdx/StackRequestManifester.java [hunk 2]
```diff
@@ -170,6 +171,9 @@ private StackV4Request setupStackRequestForCloudbreak(SdxCluster sdxCluster, Det
             setupMultiAz(sdxCluster, environment, stackRequest);
             setupGovCloud(sdxCluster, environment, stackRequest);
             stackRequest.setExternalDatabase(databaseRequestConverter.createExternalDbRequest(sdxCluster));
+            if (entitlementService.awsNativeSingleAzDatalakeEnabled(Crn.fromString(sdxCluster.getCrn()).getAccountId())) {
```
Comment (doktoric) [inline datalake/src/main/java/com/sequenceiq/datalake/service/sdx/StackRequestManifester.java:174]: should we say or multiazentitlementGranted()

**Comment by doktoric** [comment]
this logic does not cover the subnet distribution yet

**Review by akanto — APPROVED** [review]

**Comment by horadla23** [comment]
it seems cluster provision without multiAz flag already makes it possible to put every node into one AZ, so it seems subnet distribution logic looks good as is

File: core/src/main/java/com/sequenceiq/distrox/v1/distrox/converter/DistroXV1RequestToStackV4RequestConverter.java (hunk 1 above)
Comment (horadla23) [reply core/src/main/java/com/sequenceiq/distrox/v1/distrox/converter/DistroXV1RequestToStackV4RequestConverter.java]: how multi AZ comes into picture here? here we need to enable native even if multiAz not enabled

File: datalake/src/main/java/com/sequenceiq/datalake/service/sdx/StackRequestManifester.java (hunk 2 above)
Comment (horadla23) [reply datalake/src/main/java/com/sequenceiq/datalake/service/sdx/StackRequestManifester.java]: same here

**Pull request merged**